### PR TITLE
feat: implement public Push APIs and generate swagger definition document for push

### PR DESCRIPTION
**Pull request by hvn2k1**
Part of https://github.com/bucketeer-io/bucketeer/issues/904

- These Push APIs are implemented with documentation:
  - CreatePush
  - ListPush
  - GetPush
  - UpdatePush
- Refactor publish feature apis to another file

### CODE REVIEW

File: pkg/push/api/api.go
Comment (cre8ivejp) [inline pkg/push/api/api.go:1]: The new update API must not use the command layer. So, we must have an implementation similar to this when updating the proto message.
https://github.com/bucketeer-io/bucketeer/blob/main/pkg/feature/domain/feature.go#L958-L1043

File: pkg/api/api/api_grpc.go [hunk 1]
```diff
@@ -1077,6 +1082,180 @@ func (s *grpcGatewayService) UpdateFeature(
...
+			Tags: req.AddPushTags,
+		}
+	}
+
+	var deletePushTagsCommand *pushproto.DeletePushTagsCommand
+	if len(req.DeletePushTags) > 0 {
+		deletePushTagsCommand = &pushproto.DeletePushTagsCommand{
+			Tags: req.DeletePushTags,
+		}
+	}
```
[the remark this answers is not in the document]
Comment (hvn2k1) [reply pkg/api/api/api_grpc.go:1231]: I have created new pull request to support no command for push web APIs https://github.com/bucketeer-io/bucketeer/pull/1298

File: pkg/api/api/api_grpc.go [hunk 2]
```diff
@@ -1077,6 +1082,180 @@ func (s *grpcGatewayService) UpdateFeature(
 	}, nil
 }
 
+func (s *grpcGatewayService) ListPushes(
```
[the remark this answers is not in the document]
Comment (hvn2k1) [reply pkg/api/api/api_grpc.go:1085]: Yeah that's a good idea, let me refactor this part

File: pkg/api/api/api_grpc.go [hunk 3]
```diff
@@ -1077,6 +1082,180 @@ func (s *grpcGatewayService) UpdateFeature(
...
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// For security reasons we remove the service account from the API response
+	for _, p := range pushes.Pushes {
+		p.FcmServiceAccount = ""
+	}
```
[the remark this answers is not in the document]
Comment (hvn2k1) [reply pkg/api/api/api_grpc.go:1120]: I see, I have undo this

File: pkg/push/api/api.go
Comment (hvn2k1) [reply pkg/push/api/api.go:1]: Understand, I have implement the no command support in web svc in this pr: https://github.com/bucketeer-io/bucketeer/pull/1298

File: pkg/push/api/api.go [hunk 4]
```diff
@@ -561,7 +569,136 @@ func (s *PushService) UpdatePush(
...
+		}
+		prev := &domain.Push{}
+		if err = copier.Copy(prev, push); err != nil {
+			return err
+		}
+		push.Name = req.Name
+		push.Tags = req.Tags
+		updatedPushPb = push.Push
+
+		updatePushNameEvent, err = domainevent.NewEvent(
```
Comment (cre8ivejp) [inline pkg/push/api/api.go:625]: We can't have more than one event being reported per request.

Reference:
https://github.com/bucketeer-io/bucketeer/blob/6dccbca812e60f9ae5f1250906a24f300646c209/pkg/feature/api/feature.go#L792-L806

File: pkg/api/api/push.go [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+			OrderBy:              req.OrderBy,
+			OrderDirection:       req.OrderDirection,
+			SearchKeyword:        req.SearchKeyword,
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+	if res == nil {
+		s.logger.Error("Push not found")
```
Comment (Ubisoft-potato) [inline pkg/api/api/push.go:60]: We should log error like this:
```go
s.logger.Error("Failed to check CreatePush request",
			log.FieldsFromImcomingContext(ctx).AddFields(
				zap.Error(err),
				zap.String("name", req.Name),
			)...,
		)
```

**Review by Ubisoft-potato — COMMENTED** [review]
The rest code LGTM!

File: pkg/api/api/push.go (hunk 5 above)
Comment (hvn2k1) [reply pkg/api/api/push.go]: I have updated the log content

**Review by Ubisoft-potato — APPROVED** [review]
Nice work!

**Pull request merged**